### DISCUSSION TITLE
Extend functionality with the bcachefs file system

### DISCUSSION THREAD

**Discussion by Alexandero89** [jimsalterjrs/sanoid#893] [Ideas]
The short description of the repo states the following:
"Btrfs support plans are shelved unless and until btrfs becomes reliable"

And since Sanoid is a fantastic and reliable tool, I thought we could discuss extending it for Bcachefs instead of btrfs.
Because Bcachefs has been part of the mainline kernel for some time now and could well extend or replace ZFS.
And of course it would be wonderful if Sanoid could handle both filesystems in parallel

**Comment by jimsalterjrs** [comment]
bcachefs has been on my list of things to investigate more closely for a while now, but still hasn't quite bubbled up to the top. I know it supports most of the OpenZFS featureset on paper, but the one thing I haven't been able to figure out is, does it support *replication*?

Do you know?

**Reply by Alexandero89** [reply]
Good to hear.

No, I don't know about that.
But can do some research on that and come back.

**Reply by zorbathut** [reply]
"Replication" is kind of ambiguous, but as of right now, bcachefs has support for RAID, but does not have an equivalent for `zfs send`. I assume you meant the second and then the answer is "no, not yet".

It's [on their list](https://bcachefs.org/Roadmap/#Send_and_receive), they just haven't gotten to it yet.

**Reply by kimono-koans** [reply]
> but the one thing I haven't been able to figure out is, does it support _replication_?

No.

One further major hurdle/problem is there is no standard location for snapshots, ala the brtfs model.  So sanoid may define a model but other snapshot tools will also write wherever.  I have argued with Kent Overstreet about his choice of this model, and he doesn't seem to understand why it may be such a problem to have 5 different snapshot tools each with 5 different snapshot location conventions *for no real reason*.

See: https://www.reddit.com/r/zfs/comments/1eo9nqb/my_fear_is_bcachefs_will_be_the_next_btrfs/

If you would, I'd ask that you also express the why this is such an ill considered choice for a new filesystem.

Thank you.
